### PR TITLE
Kuryr configuration for portal_net and amphora VM IP ranges

### PR DESCRIPTION
**Pull request by luis5tb**

### CODE REVIEW

File: install_config/topics/ocp_osp_provisioning.adoc [hunk 1]
```diff
@@ -189,8 +189,19 @@ openshift_node_open_ports:
...
 # By default, this value is /24.
 
-# The value of openshift_portal_net in OSEv3.yml must also match the following value:
-# openshift_openstack_kuryr_service_subnet_cidr: *"172.23.192.0/18"*
+# The value of openshift_portal_net in OSEv3.yml must be within the range given
+# for openshift_openstack_kuryr_service_subnet_cidr (usually half of it),
+# ensuring it does not overlap with the range provided by
+# [openshift_openstack_kuryr_service_pool_start, openshift_openstack_kuryr_service_pool_end].
+# openshift_portal_net is the range used by services and Octavia Load Balancers
+# VIPs, while amphora VMs get Neutron ports on the range specified by
```
Comment (dulek) [inline install_config/topics/ocp_osp_provisioning.adoc:197]: Is it worth saying that those are "service" VRRP ports just for clarification?

**Comment by maxwelldb** [comment]
@luis5tb Is this related to any BZ or Jura tickets?

**Comment by luis5tb** [comment]
> @luis5tb Is this related to any BZ or Jura tickets?

no bugzilla, but this JIRA story: https://issues.redhat.com/browse/OSASINFRA-1849

File: install_config/topics/ocp_osp_provisioning.adoc [hunk 2]
```diff
@@ -189,8 +189,19 @@ openshift_node_open_ports:
 # set in the inventory file as openshift_openstack_kuryr_pod_subnet_cidr.
 # By default, this value is /24.
 
-# The value of openshift_portal_net in OSEv3.yml must also match the following value:
-# openshift_openstack_kuryr_service_subnet_cidr: *"172.23.192.0/18"*
+# The value of openshift_portal_net in OSEv3.yml must be within the range given
+# for openshift_openstack_kuryr_service_subnet_cidr (usually half of it),
```
Comment (maxwelldb) [inline install_config/topics/ocp_osp_provisioning.adoc:193]: @luis5tb Why half, and when not half?

**Review by maxwelldb — CHANGES_REQUESTED** [review]

File: install_config/topics/ocp_osp_provisioning.adoc (hunk 2 above)
Comment (luis5tb) [reply install_config/topics/ocp_osp_provisioning.adoc:193]: explanation is just below. Octavia Load Balancer require an IP (besides the VIP), so for each k8s svc, we need 2 neutron ports (1 for the amphora VM, 1 for the loadbalancer VIP, which is the same as the one the svc gets)

File: install_config/topics/ocp_osp_provisioning.adoc [hunk 3]
```diff
@@ -189,8 +189,19 @@ openshift_node_open_ports:
...
+# ensuring it does not overlap with the range provided by
+# [openshift_openstack_kuryr_service_pool_start, openshift_openstack_kuryr_service_pool_end].
+# openshift_portal_net is the range used by services and Octavia Load Balancers
+# VIPs, while amphora VMs get Neutron ports on the range specified by
+# [openshift_openstack_kuryr_service_pool_start, openshift_openstack_kuryr_service_pool_end]:
+# [OSEv4.yml]
```
Comment (maxwelldb) [inline install_config/topics/ocp_osp_provisioning.adoc:199]: This is 4 rather than 3, right? Just confirming.

**Review by maxwelldb — CHANGES_REQUESTED** [review]
One more question.

File: install_config/topics/ocp_osp_provisioning.adoc (hunk 3 above)
Comment (luis5tb) [reply install_config/topics/ocp_osp_provisioning.adoc:199]: ups... no no, that is a typo, this is OSEv3.yml

**Comment by maxwelldb** [comment]
@luis5tb What do you think of a953c21?

**Comment by maxwelldb** [comment]
@luis5tb How 'bout 3b7d2fa?

**Comment by luis5tb** [comment]
> @luis5tb How 'bout [3b7d2fa](https://github.com/openshift/openshift-docs/commit/3b7d2faa7e067231d7d96ed25cbc2342e883942b)?

looks good

**Comment by maxwelldb** [comment]
@luis5tb Cool. Do you have a QE contact who looked over this originally and might be able to +1 this PR, too?

**Comment by luis5tb** [comment]
> @luis5tb Cool. Do you have a QE contact who looked over this originally and might be able to +1 this PR, too?

This was already tested by @mohammedzee1000

**Comment by eurijon** [comment]
Hey, this change lgtm

**Comment by maxwelldb** [comment]
Thanks! @eurijon

**Comment by maxwelldb** [comment]
TODO: Squash w/co-authoring info preserved before merge.

**Review by kalexand-rh — COMMENTED** [review]
This change is probably fine, but it's not the easiest to digest.

File: install_config/topics/ocp_osp_provisioning.adoc [hunk 4]
```diff
@@ -189,8 +189,22 @@ openshift_node_open_ports:
 # set in the inventory file as openshift_openstack_kuryr_pod_subnet_cidr.
 # By default, this value is /24.
 
-# The value of openshift_portal_net in OSEv3.yml must also match the following value:
-# openshift_openstack_kuryr_service_subnet_cidr: *"172.23.192.0/18"*
+# openshift_portal_net is the range that OpenShift services and their associated Octavia
```
[the remark this answers is not in the document]
Comment (maxwelldb) [reply install_config/topics/ocp_osp_provisioning.adoc:192]: I don't really know how to ameliorate this given what the user needs to do in this process. I'm probably going to merge this and then immediately open a style fix GH issue for myself. Dev says that this change is somewhat urgent.

File: install_config/topics/ocp_osp_provisioning.adoc [hunk 5]
```diff
@@ -189,8 +189,22 @@ openshift_node_open_ports:
...
+# The value of openshift_portal_net in OSEv3.yml must be within the range that is
+# defined by openshift_openstack_kuryr_service_subnet_cidr. This range must be half
+# of openshift_openstack_kuryr_service_subnet_cidr's range. This practice ensures that
+# openshift_portal_net does not overlap with the range that load balancers' VMs use, which is
+# defined by [openshift_openstack_kuryr_service_pool_start, openshift_openstack_kuryr_service_pool_end].
+#
+# For reference only, copy the value in the next line from OSEv3.yml:
+# openshift_portal_net: *"172.30.0.0/16"*
+
+openshift_openstack_kuryr_service_subnet_cidr: *"172.30.0.0/15"*
```
[the remark this answers is not in the document]
Comment (maxwelldb) [reply install_config/topics/ocp_osp_provisioning.adoc:205]: They are, but I also don't want to have to remove them from the whole page right now. Will stick a pin in this.

File: install_config/topics/ocp_osp_provisioning.adoc [hunk 6]
```diff
@@ -189,8 +189,22 @@ openshift_node_open_ports:
 # set in the inventory file as openshift_openstack_kuryr_pod_subnet_cidr.
 # By default, this value is /24.
 
-# The value of openshift_portal_net in OSEv3.yml must also match the following value:
-# openshift_openstack_kuryr_service_subnet_cidr: *"172.23.192.0/18"*
+# openshift_portal_net is the range that OpenShift services and their associated Octavia
+# load balancer VIPs use. Amphora VMs use Neutron ports in the range that is defined by
+# [openshift_openstack_kuryr_service_pool_start, openshift_openstack_kuryr_service_pool_end].
```
[the remark this answers is not in the document]
Comment (maxwelldb) [reply install_config/topics/ocp_osp_provisioning.adoc:194]: Will fix before merging.

File: install_config/topics/ocp_osp_provisioning.adoc [hunk 7]
```diff
@@ -189,8 +189,22 @@ openshift_node_open_ports:
...
+#
+# The value of openshift_portal_net in OSEv3.yml must be within the range that is
+# defined by openshift_openstack_kuryr_service_subnet_cidr. This range must be half
+# of openshift_openstack_kuryr_service_subnet_cidr's range. This practice ensures that
+# openshift_portal_net does not overlap with the range that load balancers' VMs use, which is
+# defined by [openshift_openstack_kuryr_service_pool_start, openshift_openstack_kuryr_service_pool_end].
+#
+# For reference only, copy the value in the next line from OSEv3.yml:
```
[the remark this answers is not in the document]
Comment (maxwelldb) [reply install_config/topics/ocp_osp_provisioning.adoc:202]: > The OSEv3.yml file contains some references required by the all.yml file and all the available {product-title} cluster parameters that you can customize.

Seems like it has to exist before all.yml does.

**Review by maxwelldb — APPROVED** [review]
Okay enough. I want to do some more editing + styling in the future. See https://github.com/openshift/openshift-docs/issues/24847

**Pull request merged**